README.markdown

```diff
@@ -855,6 +856,60 @@ let sorted = mergeSort(items)  // hard to discover
 launch(&rocket)
 ```
 
+## Functions vs Computed Variables
+
+A computed variable is one that returns a calculated value each time it is accessed. That is, it doesn’t store a value. Internally it is implemented as a function.
+
+What is the difference with a function?
+
+- Semantically, a variable is state, a function is an action.
+- A function regulates access to private storage. A calculated variable may do the same in a more compact way. [Example][1].
+- A computed variable can be used with KVO, passed as a #keypath, and has facilities for observing: willSet, didSet.
+
+**You should use a variable when**
+
+- it does not throw
+- it returns a simple property
+- it doesn’t have a side effect or a verb in its name
+- it’s O(1), that is, it doesn’t incur a significant cost. In your example it will be O(n).
+- it is idempotent. Multiple identical invocations return the same value or set the object to the same state.
+
+Irrelevant reasons to prefer a variable over a function
+
+- A computed variable saves you from typing (). However, clarity is more important than brevity, so this is a weak argument.
+- A read only variable can be overriden as read/write. A function indicates it is always read only. However, Apple uses properties for read-only variables like array.count. When in doubt seek consistency with the platform.
+
+---
+The following are sources I used to write this.
```

Comment:
I suggest moving the sources to an annex at the end of the doc. In here it generates too much noise in my opinion.